artiq/test/coredevice/test_moninj.py

```diff
@@ -1,8 +1,28 @@
 import unittest
 import asyncio
+import re
+import numpy
+from contextlib import asynccontextmanager
+from unittest import IsolatedAsyncioTestCase
 
 from artiq.coredevice.comm_moninj import *
 from artiq.test.hardware_testbench import ExperimentCase
+from artiq.experiment import *
+
+
+def async_test(coro):
+    def wrapper(*args, **kwargs):
+        loop = asyncio.new_event_loop()
+        try:
+            return loop.run_until_complete(coro(*args, **kwargs))
+        finally:
+            loop.close()
+
+    return wrapper
+
+
+def get_last_integers(str):
```

Comment:
Use the channel numbers in the device instead of relying on an arbitrary device db naming scheme.

Reply:
hmm, i suppose you mean this channel number in `arguments`:
```python
device_db["ttl_urukul0_sw0"] = {
    "type": "local",
    "module": "artiq.coredevice.ttl",
    "class": "TTLOut",
    "arguments": {"channel": 0x000002}
}
```
It doesn't seem accurate so that's why I used a regex to extract the last sequence of numbers instead, or should we take account of this offset in the gateware?

Reply:
Of course it is accurate. Nothing would work if it weren't.